src/Internal/Payment/PaymentContext.php

```diff
@@ -7,9 +7,92 @@
 
 namespace WCPay\Internal\Payment;
 
+use Automattic\WooCommerce\Blocks\Payments\PaymentMethodTypeInterface;
+use WCPay\Internal\Payment\PaymentMethod\PaymentMethodInterface;
+
 /**
  * A context object, which is shared between payment states.
  */
 class PaymentContext {
```

Comment:
Do you maybe consider exposing an interface to the PaymentContext class?

Reply:
Note: The `Automattic\WooCommerce\....` dependency was not used, and I removed it in https://github.com/Automattic/woocommerce-payments/pull/7324/commits/86f09467217768a9b9ed3f141ce844f3584c5855.

> Do you maybe consider exposing an interface to the PaymentContext class?

What interfaces to use based on SOLID principles is one of the things to discuss. I, personally, wouldn't start with the `PaymentContext` because as a DTO-like object, it's got too many methods that would need to be defined in the interface.